libraries/chain/include/eosio/chain/trace.hpp

```diff
@@ -19,6 +19,30 @@ namespace eosio::chain {
    struct transaction_trace;
    using transaction_trace_ptr = std::shared_ptr<transaction_trace>;
 
+   struct sync_call_trace {
+      sync_call_trace(uint32_t sender_ordinal, account_name sender, account_name receiver, uint64_t flags, std::span<const char> data)
+         : sender_ordinal(sender_ordinal)
+         , sender(sender)
+         , receiver(receiver)
+         , flags(flags)
+         , data(data.begin(), data.end())
+      {
+      }
+
+      uint32_t                      ordinal = 1;
+      const uint32_t                sender_ordinal = 0;
+      const account_name            sender;
+      const account_name            receiver;
+      const uint64_t                flags = 0;
```

Comment:
Does it make any sense to "unwrap" the flags in the user visible trace? For example instead of a consumer just seeing `flags: 1` or `flags: 2` and needing to know what the magic numbers are, they'd see for example
```
   "read_only": false,
   "some_other_flag": true
```

Reply:
Good idea. I will unwrap it.

Reply:
One consideration is, if we use new bits in the future, we will need to add them to the call trace. But that should be OK as it is a protocol change.

Reply:
That is true... I suppose it would be most disruptive for SHIP since it would require a whole new sync_call_trace_vx whenever a new flag is added.